backend/src/routes/users.js

```diff
@@ -1,15 +1,92 @@
+const auth = require('../auth');
 const express = require('express');
 const router = express.Router();
 const models = require('../models');
+const db = require('../db');
+const utils = require('../utils');
+const randtoken = require('rand-token');
 
-router.get('/:id', (req, res) => {
-    res.send('returning specific user ' + req.params.id);
-});
+const getUser = async (req, res) => {
+    try {
+        const phoneNumber = req.body.phoneNumber;
+        const user = await db.users.getUser(phoneNumber);
+        res.status(200).send(user);
+    } catch (err) {
+        res.status(420).send('Failed to get user: ' + err);
+    }
+};
 
-router.get('/', (req, res) => {
-    models.users.findAll({}).then(function(users) {
+const getAllUsers = async (req, res) => {
+    try {
+        const users = await db.users.getAllUsers();
         res.status(200).send(users);
-    });
-});
+    } catch (err) {
+        res.status(420).send('Failed to get user list: ' + err);
+    }
+};
+
+const updateUser = async (req, res) => {
+    try {
+        const user = req.body;
+        await db.users.updateUser(user);
+        res.status(200).send('Updated user details');
+    } catch (err) {
+        res.status(420).send('Failed to update user details: ' + err);
+    }
+};
+
+const deleteUser = async (req, res) => {
+    try {
+        const phoneNumber = req.body.phoneNumber;
+        await db.users.deleteUser(phoneNumber);
+        res.status(200).send('Deleted user');
+    } catch (err) {
+        res.status(420).send('Failed to delete user: ' + err);
+    }
+};
+
+const createUser = async (req, res) => {
+    try {
+        const user = req.body;
+        const token = randtoken.generate(16);
+        user.token = token;
+        await db.users.createUser(user);
+        res.status(200).send(user);
+    } catch (err) {
+        res.status(420).send('Failed to create new user: ' + err);
+    }
+};
+
+const login = async (req, res) => {
+    try {
+        const phoneNumber = req.body.phoneNumber;
+        const pin = req.body.pin;
+        const user = await db.users.getUserByPhoneAndPin(phoneNumber, pin);
+        if (user) {
+            res.status(200).send(user);
+        } else {
+            res.status(401).send('Phone number or pin is incorrect!.');
+        }
+    } catch (err) {
+        res.status(420).send('Failed to login: ' + err.message);
+    }
+};
+
+router.post('/', utils.users.validateUserObj, createUser);
+router.get('/', auth.requireAuth, getAllUsers);
+router.put('/', auth.requireAuth, utils.users.validateUserObj, updateUser);
```

Comment:
Usually when we are updating the current user the path is something like `/users/me`